pkg/ottl/e2e/e2e_test.go

```diff
@@ -869,71 +870,108 @@ func Test_e2e_converters(t *testing.T) {
 			},
 		},
 		{
 			name:      "Using enums",
-			statement: `set(severity_number, SEVERITY_NUMBER_TRACE2) where severity_number == SEVERITY_NUMBER_TRACE`,
+			statement: []string{`set(severity_number, SEVERITY_NUMBER_TRACE2) where severity_number == SEVERITY_NUMBER_TRACE`},
 			want: func(tCtx ottllog.TransformContext) {
 				tCtx.GetLogRecord().SetSeverityNumber(2)
 			},
 		},
 		{
 			name:      "Using hex",
-			statement: `set(attributes["test"], "pass") where trace_id == TraceID(0x0102030405060708090a0b0c0d0e0f10)`,
+			statement: []string{`set(attributes["test"], "pass") where trace_id == TraceID(0x0102030405060708090a0b0c0d0e0f10)`},
 			want: func(tCtx ottllog.TransformContext) {
 				tCtx.GetLogRecord().Attributes().PutStr("test", "pass")
 			},
 		},
 		{
 			name:      "where clause without comparator",
-			statement: `set(attributes["test"], "pass") where IsMatch(body, "operation[AC]")`,
+			statement: []string{`set(attributes["test"], "pass") where IsMatch(body, "operation[AC]")`},
 			want: func(tCtx ottllog.TransformContext) {
 				tCtx.GetLogRecord().Attributes().PutStr("test", "pass")
 			},
 		},
 		{
 			name:      "where clause with Converter return value",
-			statement: `set(attributes["test"], "pass") where body == Concat(["operation", "A"], "")`,
+			statement: []string{`set(attributes["test"], "pass") where body == Concat(["operation", "A"], "")`},
 			want: func(tCtx ottllog.TransformContext) {
 				tCtx.GetLogRecord().Attributes().PutStr("test", "pass")
 			},
 		},
 		{
 			name:      "composing functions",
-			statement: `merge_maps(attributes, ParseJSON("{\"json_test\":\"pass\"}"), "insert") where body == "operationA"`,
+			statement: []string{`merge_maps(attributes, ParseJSON("{\"json_test\":\"pass\"}"), "insert") where body == "operationA"`},
 			want: func(tCtx ottllog.TransformContext) {
 				tCtx.GetLogRecord().Attributes().PutStr("json_test", "pass")
 			},
 		},
+		{
+			name:      "where clause with Contains return value",
+			statement: []string{`set(attributes["test"], "pass") where Contains(["hello", "world"], "hello")`},
+			want: func(tCtx ottllog.TransformContext) {
+				tCtx.GetLogRecord().Attributes().PutStr("test", "pass")
+			},
+		},
+		{
+			name:      "where clause with Contains ints return value",
+			statement: []string{`set(attributes["test"], "pass") where Contains([1, 2, 3, 4], 4)`},
+			want: func(tCtx ottllog.TransformContext) {
+				tCtx.GetLogRecord().Attributes().PutStr("test", "pass")
+			},
+		},
+		{
+			name:      "where clause with Contains floats return value",
+			statement: []string{`set(attributes["test"], "pass") where Contains([1.1, 2.2, 3.3, 4.4], 4.4)`},
+			want: func(tCtx ottllog.TransformContext) {
+				tCtx.GetLogRecord().Attributes().PutStr("test", "pass")
+			},
+		},
+		{
+			name: `set attribute when tag "staging" is in tags attributes slice using Contains`,
+			statement: []string{
+				`set(attributes["tags"], ["staging", "hello", "world", "work"])`,
```

Comment:
It's not clear to me why we need to support a slice of statements in this test. Isn't this first statement just a form of setup? Could it be moved into `constructLogTransformContext` in a way that doesn't interfere with the other tests?

Reply:
Yes, instead of adding new attributes in the `constructLogTransformContext` function, I simply reuse the already created slice, so the effect is the same, but I don't have to add new attributes to the current code.